spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulLifecycle.java

```diff
@@ -170,7 +173,7 @@ private void deregister(String serviceId) {
 
 	@Override
 	protected boolean isEnabled() {
-		return properties.isEnabled();
+		return lifecycleProperties.isEnabled();
```

Comment:
As I see if `spring.cloud.enabled=false` or `spring.cloud.consul.discovery.enabled=false` then  `ConsulDiscoveryClientConfiguration` will not enabled according to `@ConditionalOnConsulEnabled` and `@ConditionalOnProperty` annotations and then `ConsulLifecycle` will be eliminated despite the value of `spring.cloud.consul.discovery.lifecycle`.